ansible_collections/arista/avd/roles/eos_designs/python_modules/mlag/__init__.py

```diff
@@ -0,0 +1,361 @@
+            ip_address = get(self._hostvars, "switch.mlag_l3_ip", required=True)
+            l3_vlan_interface["ip_address"] = f"{ip_address}/31"
+
+        l3_vlan_interface["mtu"] = self._p2p_uplinks_mtu
+        l3_vlan_interface.update(l3_cfg)
+
+        # Assembling the interface dict to retain legacy order from Jinja templates.
+        return {
+            l3_vlan_interface_name: l3_vlan_interface,
+            main_vlan_interface_name: main_vlan_interface,
+        }
+
+    @cached_property
+    def port_channel_interfaces(self):
+        """
+        Return dict with one Port Channel Interface used for MLAG Peer Link
+        """
+
+        port_channel_interface_name = f"Port-Channel{self._mlag_port_channel_id}"
+        port_channel_interface = {}
+        port_channel_interface["description"] = self._avd_interface_descriptions.mlag_port_channel_interfaces()
+        port_channel_interface["type"] = "switched"
+        port_channel_interface["shutdown"] = False
+        port_channel_interface["vlans"] = get(self._hostvars, "switch.mlag_peer_link_allowed_vlans", required=True)
+        port_channel_interface["mode"] = "trunk"
+        if (p2p_uplinks_qos_profile := get(self._hostvars, "p2p_uplinks_qos_profile")) is not None:
+            port_channel_interface["service_profile"] = p2p_uplinks_qos_profile
+
+        port_channel_interface["trunk_groups"] = []
+        if self._mlag_l3 is True:
+            # Add LEAF_PEER_L3 even if we reuse the MLAG trunk group for underlay peering
+            # since this trunk group is also used for overlay iBGP peerings
+            port_channel_interface["trunk_groups"].append(self._trunk_groups_mlag_l3_name)
+
+        port_channel_interface["trunk_groups"].append(self._trunk_groups_mlag_name)
+
+        if (struct_cfg := get(self._hostvars, "switch.mlag_port_channel_structured_config")) is not None:
+            port_channel_interface["struct_cfg"] = struct_cfg
+
+        return {port_channel_interface_name: port_channel_interface}
+
+    @cached_property
+    def ethernet_interfaces(self):
+        """
+        Return dict with Ethernet Interfaces used for MLAG Peer Link
+        """
+
+        mlag_interfaces = get(self._hostvars, "switch.mlag_interfaces", [])
+        if not mlag_interfaces:
+            return None
+
+        ethernet_interfaces = {}
+        for mlag_interface in mlag_interfaces:
+            ethernet_interface = {}
+            ethernet_interface["peer"] = self._mlag_peer
+            ethernet_interface["peer_interface"] = mlag_interface
+            ethernet_interface["peer_type"] = "mlag_peer"
+            ethernet_interface["description"] = self._avd_interface_descriptions.mlag_ethernet_interfaces(mlag_interface)
+            ethernet_interface["type"] = "switched"
+            ethernet_interface["shutdown"] = False
+            ethernet_interface["channel_group"] = {
+                "id": self._mlag_port_channel_id,
+                "mode": "active",
+            }
```

Comment:
Wondering if this kind of syntax would not make the code more readable (could apply in other places):

```suggestion
            ethernet_interface = {
                "peer": self._mlag_peer,
                "peer_interface": mlag_interface,
                "peer_type": "mlag_peer",
                "description": self._avd_interface_descriptions.mlag_ethernet_interfaces(mlag_interface),
                "type": "switched",
                "shutdown": False,
                "channel_group": {
                    "id": self._mlag_port_channel_id,
                    "mode": "active",
                },
             }
```

Reply:
fixed